rampwf/prediction_types/multiclass.py

```diff
@@ -12,61 +12,57 @@
-        shape = self.y_proba.shape
-        if len(shape) != 2:
-            raise ValueError('Multiclass y_proba should be 2-dimensional, '
-                             'instead it is {}-dimensional'.format(len(shape)))
-        # if shape[1] != len(labels):
-        #    raise ValueError('Vectors in multiclass y_proba should be '
-        #                     '{}-dimensional, instead they are {}-dimensional'.
-        #                     format(len(labels), shape[1]))
 
-    def _init_from_pred_labels(self, y_pred_labels):
-        type_of_label = type(self.labels[0])
-        self.y_proba = np.zeros(
-            (len(y_pred_labels), len(self.labels)), dtype=np.float64)
-        for ps_i, label_list in zip(self.y_proba, y_pred_labels):
-            # converting single labels to list of labels, assumed below
-            if type(label_list) != np.ndarray and type(label_list) != list:
-                label_list = [label_list]
-            label_list = list(map(type_of_label, label_list))
-            for label in label_list:
-                ps_i[self.labels.index(label)] = 1.0 / len(label_list)
+def _init_from_pred_labels(self, y_pred_labels):
+    """Initalize y_pred to uniform for labels in y_pred_labels.
 
-    def set_valid_in_train(self, predictions, test_is):
-        self.y_proba[test_is] = predictions.y_proba
+    Initialize multiclass Predictions from ground truth. y_pred_labels
+    can be a single (positive) label in which case the corresponding
+    column gets probability of 1.0. In the case of multilabel (k > 1
+    positive labels), the columns corresponing the positive labels
+    get probabilities 1/k.
 
-    @property
-    def valid_indexes(self):
-        return ~np.isnan(self.y_proba[:, 0])
+    Parameters
+    ----------
+    y_pred_labels : object or list of objects (of the same type)
+    """
+    type_of_label = type(self.label_names[0])
+    self.y_pred = np.zeros(
+        (len(y_pred_labels), len(self.label_names)), dtype=np.float64)
+    for ps_i, label_list in zip(self.y_pred, y_pred_labels):
+        # converting single labels to list of labels, assumed below
+        if type(label_list) != np.ndarray and type(label_list) != list:
+            label_list = [label_list]
+        label_list = list(map(type_of_label, label_list))
+        for label in label_list:
+            ps_i[self.label_names.index(label)] = 1.0 / len(label_list)
 
-    @property
-    def y_pred(self):
-        return self.y_proba
 
-    @property
-    def y_pred_label_index(self):
-        """Multi-class y_pred is the index of the predicted label."""
-        return np.argmax(self.y_proba, axis=1)
+@property
+def _y_pred_label_index(self):
+    """Multi-class y_pred is the index of the predicted label."""
+    return np.argmax(self.y_pred, axis=1)
 
-    @property
-    def y_pred_label(self):
-        return self.labels[self.y_pred_label_index]
+
+def make_predictions_type(label_names=[]):
```

Comment:
I do use it for regression, e.g. in epidemium, where there are 6 targets. My semantics is that label_names name the columns of y_pred, when y_pred is 2D. It is [] when it is 1D. So len(label_names) is the second dim of y_pred.